packages/protobuf/src/private/reflect.ts

```diff
@@ -60,7 +60,12 @@ export function clearField(
   if (field.repeated) {
     target[localName] = [];
   } else if (field.oneof) {
-    target[field.oneof.localName] = { case: undefined };
+    if (
+      (target[field.oneof.localName] as { case: string }).case ===
+      field.localName
+    ) {
+      target[field.oneof.localName] = { case: undefined };
+    }
```

Comment:
This internal function was added in https://github.com/bufbuild/protobuf-es/pull/702. It used to always reset a `oneof`. This wasn't an issue with how we used it, but now that it's a public API, it seems to make much more sense to only reset a `oneof` if the given field was actually selected.